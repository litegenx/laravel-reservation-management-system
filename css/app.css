.router-link-active {
  cursor: pointer;
}

