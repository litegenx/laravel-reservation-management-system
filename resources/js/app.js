require('./bootstrap');
import 'babel-polyfill';
import Vue from 'vue';
import router from './router';
import store from './store';
import App from './components/App';
import vuetify from './plugins/vuetify';

new Vue({
    el: '#app',
    store,
    router,
    components: { App },
    template: '<App/>',
    vuetify,
});
