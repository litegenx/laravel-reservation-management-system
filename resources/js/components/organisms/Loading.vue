<template>
    <v-layout
        row
        justify-center
    >
        <v-dialog
            :value="isActiveOverlay"
            persistent
            content
            content-class="centered-dialog"
        >
            <v-container fill-height>
                <v-layout
                    column
                    justify-center
                    align-center
                >
                    <v-progress-circular
                        indeterminate
                        :size="70"
                        :width="7"
                        :color="progressColor"
                    />
                    <h1 v-if="message != null">
                        {{ message }}
                    </h1>
                </v-layout>
            </v-container>
        </v-dialog>
    </v-layout>
</template>

<script>
    import { mapGetters } from 'vuex';

    export default {
        computed: {
            ...mapGetters({
                isActiveOverlay: 'loading/isActiveOverlay',
                message: 'loading/getMessage',
                progressColor: 'loading/getProgressColor',
            }),
        },
    };
</script>

<style>
    .v-dialog.centered-dialog {
        background: #282c2dad;
        box-shadow: none;
        border-radius: 6px;
        width: auto;
        color: whitesmoke;
    }
</style>
