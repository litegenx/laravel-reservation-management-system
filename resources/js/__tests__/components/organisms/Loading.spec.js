import { mount } from '@vue/test-utils';
import Vuex from 'vuex';
import Loading from '../../../components/organisms/Loading';
import app from '../app';
import setupLocalVue from '../setupLocalVue';

describe('Loading', () => {
    it('should not render loading if not active', () => {
        const wrapper = mount(app(Loading, 'bottom'), setupLocalVue({
            store: new Vuex.Store({
                getters: {
                    'loading/isActiveOverlay': () => false,
                    'loading/getMessage': () => '',
                    'loading/getProgressColor': () => undefined,
                },
            }),
        }));
        expect(wrapper.element).toMatchSnapshot('not active');

        expect(wrapper.isVueInstance()).toBeTruthy();
        expect(wrapper.findAll('.v-overlay__content')).toHaveLength(0);
    });

    it('should render loading', () => {
        const wrapper = mount(app(Loading, 'bottom'), setupLocalVue({
            store: new Vuex.Store({
                getters: {
                    'loading/isActiveOverlay': () => true,
                    'loading/getMessage': () => '',
                    'loading/getProgressColor': () => '#033',
                },
            }),
        }));
        expect(wrapper.element).toMatchSnapshot('active');

        expect(wrapper.isVueInstance()).toBeTruthy();
        expect(wrapper.findAll('.v-overlay__content')).toHaveLength(1);
        expect(wrapper.find('.v-overlay__content').text()).toBe('');
    });

    it('should render loading message', () => {
        const wrapper = mount(app(Loading, 'bottom'), setupLocalVue({
            store: new Vuex.Store({
                getters: {
                    'loading/isActiveOverlay': () => true,
                    'loading/getMessage': () => 'test message',
                    'loading/getProgressColor': () => undefined,
                },
            }),
        }));
        expect(wrapper.element).toMatchSnapshot('message');

        expect(wrapper.isVueInstance()).toBeTruthy();
        expect(wrapper.findAll('.v-overlay__content')).toHaveLength(1);
        expect(wrapper.find('.v-overlay__content').text()).toBe('test message');
    });
});
