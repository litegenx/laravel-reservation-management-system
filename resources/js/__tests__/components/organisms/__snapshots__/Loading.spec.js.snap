// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should not render loading if not active: not active 1`] = `
<div
  class="v-application v-application--is-ltr theme--light"
  data-app="true"
  id="app"
>
  <div
    class="v-application--wrap"
  >
    <main
      app="true"
      class="v-content"
      classname="mb-5"
      style="padding: 0px 0px 0px 0px;"
    >
      <div
        class="v-content__wrap"
      >
        <div
          id="test-contents"
        >
          contents
        </div>
      </div>
    </main>
    <div
      class="v-overlay text-center theme--dark"
      style="z-index: 5;"
    >
      <div
        class="v-overlay__scrim"
        style="opacity: 0; background-color: rgb(33, 33, 33); border-color: #212121;"
      />
    </div>
  </div>
</div>
`;

exports[`Loading should render loading message: message 1`] = `
<div
  class="v-application v-application--is-ltr theme--light"
  data-app="true"
  id="app"
>
  <div
    class="v-application--wrap"
  >
    <main
      app="true"
      class="v-content"
      classname="mb-5"
      style="padding: 0px 0px 0px 0px;"
    >
      <div
        class="v-content__wrap"
      >
        <div
          id="test-contents"
        >
          contents
        </div>
      </div>
    </main>
    <div
      class="v-overlay text-center v-overlay--active theme--dark"
      style="z-index: 5;"
    >
      <div
        class="v-overlay__scrim"
        style="opacity: 0.46; background-color: rgb(33, 33, 33); border-color: #212121;"
      />
      <div
        class="v-overlay__content"
      >
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          class="v-progress-circular v-progress-circular--indeterminate"
          role="progressbar"
          style="height: 64px; width: 64px;"
        >
          <svg
            style="transform: rotate(0deg);"
            viewBox="21.333333333333332 21.333333333333332 42.666666666666664 42.666666666666664"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              class="v-progress-circular__overlay"
              cx="42.666666666666664"
              cy="42.666666666666664"
              fill="transparent"
              r="20"
              stroke-dasharray="125.664"
              stroke-dashoffset="125.66370614359172px"
              stroke-width="2.6666666666666665"
            />
          </svg>
          <div
            class="v-progress-circular__info"
          />
        </div>
         
        <div
          class="mt-3 font-weight-bold title"
        >
          
        test message
    
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Loading should render loading: active 1`] = `
<div
  class="v-application v-application--is-ltr theme--light"
  data-app="true"
  id="app"
>
  <div
    class="v-application--wrap"
  >
    <main
      app="true"
      class="v-content"
      classname="mb-5"
      style="padding: 0px 0px 0px 0px;"
    >
      <div
        class="v-content__wrap"
      >
        <div
          id="test-contents"
        >
          contents
        </div>
      </div>
    </main>
    <div
      class="v-overlay text-center v-overlay--active theme--dark"
      style="z-index: 5;"
    >
      <div
        class="v-overlay__scrim"
        style="opacity: 0.46; background-color: rgb(33, 33, 33); border-color: #212121;"
      />
      <div
        class="v-overlay__content"
      >
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          class="v-progress-circular v-progress-circular--indeterminate"
          role="progressbar"
          style="height: 64px; width: 64px; color: rgb(0, 51, 51); caret-color: #033;"
        >
          <svg
            style="transform: rotate(0deg);"
            viewBox="21.333333333333332 21.333333333333332 42.666666666666664 42.666666666666664"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              class="v-progress-circular__overlay"
              cx="42.666666666666664"
              cy="42.666666666666664"
              fill="transparent"
              r="20"
              stroke-dasharray="125.664"
              stroke-dashoffset="125.66370614359172px"
              stroke-width="2.6666666666666665"
            />
          </svg>
          <div
            class="v-progress-circular__info"
          />
        </div>
         
        <!---->
      </div>
    </div>
  </div>
</div>
`;
